panc/src/main/scripts/panlint

```diff
@@ -0,0 +1,287 @@
+#!/usr/bin/env python
+# coding=utf8
+
+import re
+import argparse
+from glob import glob
+from colorama import Fore, Style, init as colorama_init
+from sys import stdout, exit as sys_exit
+from inspect import getmembers, ismethod
+from prettytable import PrettyTable
+
+RS_COMMENT = r'(?:#.*|@{.*})$'
+
+RE_STRING = re.compile(r'''('.*?'|".*?"(?<!\\"))''')
+
+RE_FIRST_LINE = re.compile(r'^(?:(?:declaration|unique|structure|object) )?template (?:(?:[\w-]+/)+)?[\w-]+;$')
+RE_COMMENT_LINE = re.compile(r'^\s*' + RS_COMMENT)
+RE_TRAILING_COMMENT = re.compile(r'\s*' + RS_COMMENT)
+RE_ANNOTATION = re.compile(r'@{(?:.|\n)*?}')
+
+# Simple regular-expression based checks that will be performed against all non-ignored lines
+# Every pattern must provide a single capturing group named "error"
+LINE_PATTERNS = {
+    "Indentation should be a multiple of four spaces" : re.compile(r'^(?!(    )*(\S|$))(?P<error>\s*)'),
+    "Spaces should be used instead of tabs" : re.compile(r'(?P<error>\t+)'),
+    "Trailing whitespace" : re.compile(r'(?P<error>\s+$)'),
+    "Use dict() instead of nlist()" : re.compile(r'(?P<error>nlist)'),
+    "Include statements no longer need curly braces" : re.compile(r'''include\s+(?P<error>{.+})'''),
```

Comment:
hmmm, you can try, but how complicated do we allow single line dml to be? i would start with `[^{]+`, i.e. only simple things.